src/SpaceStation.cpp

```diff
@@ -355,6 +367,27 @@ bool SpaceStation::GetDockingClearance(Ship *s, std::string &outMsg)
 			sd.maxOffset = std::max((pPort->maxShipSize / 2 - bboxRad), float(pPort->maxShipSize / 5.0));
 			sd.maxOffset *= sd.maxOffset;
 			outMsg = stringf(Lang::CLEARANCE_GRANTED_BAY_N, formatarg("bay", i + 1));
+			if (this->IsGroundStation()) {
```

Comment:
I would prefer that this logic be moved into Lua and triggered with an `onDockingClearanceGranted` / `onDockingClearanceDenied` event. This reduces code duplication and moves the code that's merely concerned with pretty-printing the UI away from the code that's responsible for determining whether docking is even possible.

Reply:
Okay, I'll see what I can do.